### PR TITLE
Operation scripting for recording and journaling events

### PR DESCRIPTION
**Pull request by xbrianh**
This adds commands to the data-store operations scripting for events:
- record: record events for a list of keys, or for every bundle in replica
- journal: compile flashflood events into journals
- update: apply event updates to the event journals

### CODE REVIEW

File: dss/operations/events.py [hunk 1]
```diff
@@ -25,12 +35,86 @@
                                          nargs="*",
                                          help="bundle keys to generate documents.")})
 def bundle_metadata_document(argv: typing.List[str], args: argparse.Namespace):
+    replica = Replica[args.replica]
     for key in args.keys:
```
[the remark this answers is not in the document]
Comment (xbrianh) [reply dss/operations/events.py]: fixed

File: dss/operations/events.py [hunk 2]
```diff
@@ -21,16 +31,77 @@
...
+            print(msg)
+
+@events.action("journal",
+               arguments={"--prefix": dict(required=True,
+                                           help="flashflood prefix to journal events"),
+                          "--minimum-number-of-events": dict(default=100),
+                          "--minimum-size": dict(default=1024 * 1024)})
+def journal(argv: typing.List[str], args: argparse.Namespace):
+    ff = FlashFlood(resources.s3, Config.get_flashflood_bucket(), args.prefix)
+    ff.journal(int(args.number_of_events))
```
Comment (DailyDreaming) [inline dss/operations/events.py]: Should `args.number_of_events` be `args.minimum_number_of_events`?  Also, is `args.minimum_size` used?

File: dss/operations/events.py [hunk 3]
```diff
@@ -21,16 +31,77 @@
...
+                    else:
+                        if tombstone_id and tombstone_id in key:
+                            continue
+                        else:
+                            tombstone_id = None
+                            sqsm.send(cmd_template.format(key))
+
+        handle = Config.get_blobstore_handle(replica)
+        map_bucket(forward_keys, handle, replica.bucket, f"bundles/")
+        monitor_logs(logs, job_id, start_time)
```
Comment (DailyDreaming) [inline dss/operations/events.py]: Should this be `args.job_id`?  Also, this defaults to `None` and that string will be used as the filter pattern in `monitor_logs()`.  Not sure if that's intentional.

File: dss/operations/events.py [hunk 4]
```diff
@@ -21,16 +31,77 @@
...
+    else:
+        for key in args.keys:
+            msg = json.dumps(dict(action="record event", job_id=job_id, replica=replica.name, key=key))
```
Comment (DailyDreaming) [inline dss/operations/events.py]: Again, should this be `args.job_id`?

File: dss/operations/events.py (hunk 2 above)
Comment (xbrianh) [reply dss/operations/events.py]: fixed

File: dss/operations/events.py (hunk 4 above)
Comment (xbrianh) [reply dss/operations/events.py]: Nope, `job_id` is assigned on line 60

File: daemons/dss-operations/.chalice/config.json [hunk 5]
```diff
@@ -7,8 +7,54 @@
   "stages": {
     "dev": {
       "api_gateway_stage": "dev",
-      "tags": {},
-      "environment_variables": {}
+      "tags": {
```
Comment (DailyDreaming) [inline daemons/dss-operations/.chalice/config.json]: Leftovers.

**Pull request merged**